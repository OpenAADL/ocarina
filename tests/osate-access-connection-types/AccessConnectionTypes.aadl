data d
end d;

data implementation d.a
end d.a;

data implementation d.b
end d.b;

data DB
  features 
    shared: provides data access d.a;
end DB;

process P
  features
    needs: requires data access d.b;
end P;

system S
end S;

system implementation S.basic
  subcomponents
    theDB: data DB;
    proc: process P;
  connections
    -- bad: data classifiers of the two features are not identical
    data access theDB.shared -> proc.needs;
end S.basic;
