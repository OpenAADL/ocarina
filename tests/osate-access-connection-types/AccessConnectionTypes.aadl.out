AccessConnectionTypes.aadl:29:05: theDB.shared (entity reference)  and proc.needs (entity reference)  do not have compatible types
Cannot analyze AADL specifications
